src/Microsoft.AspNetCore.BuildTools.ApiCheck/ApiListingGenerator.cs

```diff
@@ -151,31 +159,39 @@ private static TypeKind GetTypeKind(TypeInfo type)
             return TypeKind.Unknown;
         }
 
-        private static IEnumerable<GenericParameterDescriptor> GetGenericConstraintsFor(IEnumerable<TypeInfo> genericArguments)
+        private static IEnumerable<GenericParameterDescriptor> GetGenericConstraintsFor(TypeInfo[] genericArguments)
         {
-            foreach (var typeArgument in genericArguments)
+            for (var i = 0; i < genericArguments.Length; i++)
             {
+                var typeArgument = genericArguments[i];
                 var constraintDescriptor = new GenericParameterDescriptor();
                 constraintDescriptor.Source = typeArgument;
 
-                if (typeArgument.BaseType != null &&
-                    typeArgument.BaseType != typeof(object)
-                    && typeArgument.BaseType != typeof(ValueType))
+                if (typeArgument.IsGenericParameter)
                 {
-                    constraintDescriptor.BaseTypeOrInterfaces.Add(TypeDescriptor.GetTypeNameFor(typeArgument.BaseType.GetTypeInfo()));
-                }
+                    if (typeArgument.BaseType != null &&
+                        typeArgument.BaseType != typeof(object)
```

Comment:
`typeArgument.BaseType != typeof(object) &&` (to be consistent with the previous line)